autoscheduler/scraper/management/commands/utils/pdf_parser.py

```diff
@@ -0,0 +1,145 @@
+    """ Parses a single page of a PDF, extracting a list of grade data for each section.
+
+        Args:
+            page_obj: A PyPDF2.pdf.PageObject representing the current page
+        Returns:
+            A list of GradeData objects, with the GPA as None
+    """
+
+    text = sanitize_page(page_obj) # Splits the text into separate lines
+    i = 0 # The current row in the pdf we're on
+    grade_data = [] # list of GradeData objects
+
+    while i < len(text):
+        skip_count = get_pdf_skip_count(text[i])
+        old_pdf_style = skip_count[0]
+        count = skip_count[1] # The actual count of how many rows should be skipped
+
+        if count != -1:
+            i += count
+        else: # These are the rows that have actual grade data on them
+            # Split the row into separate sections
+            section_row = text[i : i + LEN_SECTION_ROW]
+            try:
+                dept, course_num, section_num = section_row[0].split("-")
+
+                letter_grades = extract_letter_grades(section_row, old_pdf_style)
+
+                grade = GradeData(dept, course_num, section_num, letter_grades, None)
+
+                grade_data.append(grade)
+
+                i += LEN_SECTION_ROW
+            except ValueError: # When would this happen? What causes it?
+                i += LEN_SECTION_ROW - 1
+
+    return grade_data
+
+def calculate_gpa(letter_grades: Dict) -> float:
+    """ Given a series of letter grades, calculates the GPA of the section.
+
+        Args:
+            letter_grades: A list of integers representing how many students got
+                           each letter grade
+        Returns:
+            The calculated gpa.
+    """
+    weights = [4.0, 3.0, 2.0, 1.0, 0.0] # A to F GPAs, respectively
+    grades = [letter_grades[char] for char in ["A", "B", "C", "D", "F"]]
+    num_students = sum(grades)
+
+    gpa = 0.0
+    for students_with_grade, weight in zip(grades, weights):
+        gpa += students_with_grade * weight
+    return gpa / num_students
+
+def parse_pdf(pdf_path: str) -> List[GradeData]:
+    """ Loads the pdf from the given path and calls parse_page on each page, then collects
+        the results into one array.
+
+        Args:
+            pdf_path: The path to the pdf
+        Returns:
+            A list of GradeData, with each GPA calculated
+    """
+
+    with open(pdf_path, "rb") as file:
+        pdf_reader = PyPDF2.PdfFileReader(file)
+        pdf_data = []
+
+        # Iterate through all pdf pages
+        for i in range(pdf_reader.numPages):
+            # Parse the individual page
+            page_data = parse_page(pdf_reader.getPage(i))
+
+            # Calculate the GPA for each grade distribution
+            for grade in page_data:
+                grade.gpa = calculate_gpa(grade.letter_grades)
+
+            # Add this page's data to the list of GradeData to be returned
+            pdf_data += page_data

```

Comment:
```suggestion
            pdf_data.extend(page_data)
```
This should be **very** marginally faster and is more generic, so this way is preferable